value-processor/src/org/immutables/value/processor/Immutables.generator

```diff
@@ -4311,6 +4321,27 @@ private static final long serialVersionUID = [literal serialVersion];
 [/if]
 [/template]
 
+[template generateToBuilder Type type]
+public [builderReturnType type] toBuilder() {
+  return [castBuildStagedBuilder type][type.factoryBuilder.relative]()[/castBuildStagedBuilder]
+  [if type.generateBuilderFrom]
+    .[type.names.from](this);
+  [else]
+    [for v in type.settableAttributes]
+      [if v.encoding]
+    .[rr.builderCopyFrom v]([v.names.get]())
+      [else if v.nullable]
+    .[v.names.init]([v.names.get]())
+      [else if v.collectionType]
+    .[v.names.addAll]([v.names.get]())
+      [else if v.mapType]
+    .[v.names.putAll]([v.names.get]())
+      [else]
+    .[v.names.init]([v.names.get]())[/if][/for];
```

Comment:
There is small issue here. Sometimes semicolon would end up on a separate line (for example, if last attribute is a collection). I am not sure how to deal with it. For iterator has no way to check for last iteration i.e. `for.last` does not exist and can't be easily added (unless enhanced for loop in generators is replaces with iterator-and-while loop). If I missed something (this whole generator/templating framework is quite unusual to me) could you please point in the right direction.